### PR TITLE
native: add Ledger contract

### PR DESCRIPTION
**Pull request by roman-khimov**
But don't change the way we process/store transactions and blocks. Effectively
it's just an interface for smart contracts that replaces old syscalls.

Transaction definition is moved temporarily to runtime package and Block
definition is removed (till we solve #1691 properly).

### CODE REVIEW

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/core/native/ledger.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		return stackitem.Make(-1)
+	}
+	return stackitem.Make(h)
+}
+
+// getTransactionFromBlock returns transaction with the given index from the
+// block with height or hash specified.
+func (l *Ledger) getTransactionFromBlock(ic *interop.Context, params []stackitem.Item) stackitem.Item {
+	hash := getBlockHashFromItem(ic.Chain, params[0])
+	index, err := params[1].TryInteger()
```
Comment (fyrchik) [inline pkg/core/native/ledger.go:140]: `toBigInt`? Or even `toUint32`, as `MaxTransactionsPerBlock` is less than `math.MaxUint16`.

File: pkg/core/native/ledger.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	return index <= height && index+MaxTraceableBlocks > height
+}
+
+// getBlockHashFromItem converts given stackitem.Item to block hash using given
+// Blockchainer if needed. Interop functions accept both block numbers and
+// block hashes as parameters, thus this function is needed. It's supposed to
+// be called within VM context, so it panics if anything goes wrong.
+func getBlockHashFromItem(bc blockchainer.Blockchainer, item stackitem.Item) util.Uint256 {
+	bigindex, err := item.TryInteger()
+	if err == nil && bigindex.IsInt64() {
```
Comment (fyrchik) [inline pkg/core/native/ledger.go:169]: `IsUint64` and don't check for `<0` ?

**Comment by fyrchik** [comment]
Otherwise LGTM.

**Review by fyrchik — APPROVED** [review]

File: pkg/core/native/ledger.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/core/native/ledger.go:169]: `toUint32` fits precisely here, and yes, these helpers make code a bit simpler.

Comment (roman-khimov) [reply pkg/core/native/ledger.go:169]: Except we can't panic here, so I'll leave this one as is.

File: pkg/core/native/ledger.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/core/native/ledger.go]: Fixed.

**Pull request merged**